claat/parser/md/parse_test.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package md
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/googlecodelabs/tools/claat/types"
+
+	"golang.org/x/net/html"
+)
+
+func buildParserWithStep(markup string) *parserState {
+	ps := parserState{
+		tzr: html.NewTokenizer(strings.NewReader(markup)),
+		c:   &types.Codelab{},
+	}
+	ps.currentStep = ps.c.NewStep("title")
+	return &ps
+}
+
+func TestHandleCodelabTitle(t *testing.T) {
+	// Set up.
+	title := "Egret"
+	ps := buildParserWithStep(fmt.Sprintf("<h1>%s</h1>", title))
+	ps.advance()
+	handleCodelabTitle(ps)
+
+	if ps.c.Title != title {
+		t.Errorf("[%q] got %v, want %v", title, ps.c.Title, title)
+	}
+}
+
+func TestProcessDuration(t *testing.T) {
+	tests := []struct {
+		in  string
+		out time.Duration
+		err error
+	}{
+		// Test an easy case.
+		{"1:00", time.Hour, nil},
+		// Test an easy case with period delimiter.
+		{"1.00", time.Hour, nil},
+		// Test a weird number case.
+		{"13:37", 817 * time.Minute, nil},
+		// Test that minutes can be longer than an hour.
+		{"0:90", 90 * time.Minute, nil},
+		// Test zero.
+		{"0", 0, nil},
+		// Test more than two digits in the hours place.
+		{"100:00", 100 * time.Hour, nil},
+		// Test that hours must be present.
+		{":25", 0, errors.New("")},
```

Comment:
hm.. I'm trying to figure out why you need a "pseudo" empty error, which is neither `nil` nor has a string?